### PR TITLE
4747 Refactor OverlayTests

### PR DESCRIPTION
**Pull request by EvgeniiKlimenko**

### CODE REVIEW

**Review by MayaElf — APPROVED** [review]

File: jdi-light-vuetify-tests/src/main/java/io/github/com/custom/OverlayWithCloseButton.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package io.github.com.custom;
+
+import com.epam.jdi.light.vuetify.elements.common.Overlay;
+import com.epam.jdi.light.vuetify.elements.common.VuetifyButton;
+
+import static com.jdiai.tools.Timer.waitCondition;
+
+public class OverlayWithCloseButton extends Overlay {
+    private static final String CLOSE_BUTTON_PATH = ".v-overlay__content button";
```
Comment (pnatashap) [inline jdi-light-vuetify-tests/src/main/java/io/github/com/custom/OverlayWithCloseButton.java:9]: We do not need .v-overlay__content in locator, as we need to customize the content

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/common/OverlaysTests.java [hunk 2]
```diff
@@ -66,26 +56,31 @@ public void darkThemeOverlaysTest() {
         absoluteOverlayButton.show();
         absoluteOverlayButton.click();
         absoluteOverlay.has().darkTheme();
+        absoluteOverlay.close();
     }
 
     @Test(description = "Check overlay opacity value")
     public void opacityOverlaysTest() {
         opacityOverlayButton.show();
+        waitCondition(() -> opacityOverlayButton.core().isClickable());
```
Comment (pnatashap) [inline jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/common/OverlaysTests.java:65]: We can wait until Overlay is displayed, clickable looks very strange

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/common/OverlaysTests.java [hunk 3]
```diff
@@ -66,26 +56,31 @@ public void darkThemeOverlaysTest() {
...
         absoluteOverlay.is().displayed();
+        absoluteOverlay.close();
     }
 
     @Test(description = "Check overlay z-index value")
     public void zIndexOverlaysTest() {
         zIndexOverlayButton.show();
         zIndexOverlayButton.click();
         zIndexOverlay.has().zIndex(0);
+        zIndexOverlay.close();
```
Comment (pnatashap) [inline jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/common/OverlaysTests.java:84]: Missing test for Advanced Overlay

**Review by pnatashap — CHANGES_REQUESTED** [review]

File: jdi-light-vuetify-tests/src/main/java/io/github/com/custom/OverlayWithCloseButton.java (hunk 1 above)
Comment (EvgeniiKlimenko) [reply jdi-light-vuetify-tests/src/main/java/io/github/com/custom/OverlayWithCloseButton.java]: Changed locator for button.

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/common/OverlaysTests.java (hunk 2 above)
Comment (EvgeniiKlimenko) [reply jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/common/OverlaysTests.java]: Changed to wait for Overlay.isDisplayed

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/common/OverlaysTests.java (hunk 3 above)
Comment (EvgeniiKlimenko) [reply jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/common/OverlaysTests.java:84]: Test looks like this
        advancedOverlayCard.show();
        advancedOverlayCard.hover();
        advancedOverlay.has().backgroundColor(CYAN_DARKEN_5.value());
so, stars and text are related to a Card element, not Overlay. The only new thing, that appears on the overlay is a button 'SEE MORE INFO'.

File: jdi-light-vuetify-tests/src/main/java/io/github/com/custom/OverlayWithCloseButton.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import com.epam.jdi.light.vuetify.elements.common.Overlay;
+import com.epam.jdi.light.vuetify.elements.common.VuetifyButton;
+
+import static com.jdiai.tools.Timer.waitCondition;
+
+public class OverlayWithCloseButton extends Overlay {
+    private static final String CLOSE_BUTTON_PATH = "button[class*='v-btn']";
+
+    public void close() {
+        VuetifyButton closeButton = new VuetifyButton().setCore(VuetifyButton.class, find(CLOSE_BUTTON_PATH));
```
[the remark this answers is not in the document]
Comment (EvgeniiKlimenko) [reply jdi-light-vuetify-tests/src/main/java/io/github/com/custom/OverlayWithCloseButton.java:12]: I am not only want to click on a button, I also waiting until it is clickable in waitCondition().
If I delete that casting, I will need to find button by locator twice.

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/common/OverlaysTests.java [hunk 5]
```diff
@@ -66,26 +56,31 @@ public void darkThemeOverlaysTest() {
...
         opacityOverlayButton.show();
         opacityOverlayButton.click();
+        waitCondition(() -> opacityOverlay.isDisplayed());
         opacityOverlay.has().opacity(1);
+        opacityOverlay.close();
     }
 
     @Test(description = "Check that overlay is displayed or not")
     public void displayedOverlaysTest() {
         absoluteOverlay.is().hidden();
```
[the remark this answers is not in the document]
Comment (EvgeniiKlimenko) [reply jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/common/OverlaysTests.java:73]: Done

File: jdi-light-vuetify-tests/src/main/java/io/github/com/custom/OverlayWithCloseButton.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (pnatashap) [reply jdi-light-vuetify-tests/src/main/java/io/github/com/custom/OverlayWithCloseButton.java:12]: UIElement also have core(), so condition will be the same

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/common/OverlaysTests.java (hunk 3 above)
Comment (pnatashap) [reply jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/common/OverlaysTests.java:78]: Мы НЕ ТЕСТИРУЕМ фреймворк, мы показываем, как им пользоваться, мы должны показать, как описывать сложный Overlay!

**Review by pnatashap — CHANGES_REQUESTED** [review]

File: jdi-light-vuetify-tests/src/main/java/io/github/com/custom/OverlayWithCloseButton.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (EvgeniiKlimenko) [reply jdi-light-vuetify-tests/src/main/java/io/github/com/custom/OverlayWithCloseButton.java:12]: Agree, found it. Changed to UIElement closeButton = ...

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/common/OverlaysTests.java (hunk 3 above)
Comment (EvgeniiKlimenko) [reply jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/common/OverlaysTests.java:78]: I don't get it.
All those elements you mentioned above are present on Card element. For Card element we have CardsTest class. Thus, user can check this class to see how to test Card element.
And I want to emphasise, that this test was written long before I came to project by ElenaP user.

**Comment by pnatashap** [comment]
Will fix later

**Pull request merged**